File: 2022-2023/sourcecode/lab/lab2/ParcelableExampleJava/app/src/main/java/pl/edu/uwr/pum/parcelableexamplejava/SecondActivity.java
package pl.edu.uwr.pum.parcelableexamplejava;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        TextView textView = findViewById(R.id.textView);
        Intent intent = getIntent();
        Properties prop = intent.getParcelableExtra(MainActivity.EXTRA_KEY);
        SerializableProperties serProp = (SerializableProperties) intent.getSerializableExtra(MainActivity.EXTRA_SERIALIZABLE);

        String result = "";

        if(prop != null)
            result = prop.getA() + prop.getB() + " " + prop.getC() + "---" + serProp.getA() +
                    serProp.getB() + " " + serProp.getC();

        textView.setText(result);
    }
}

File: 2022-2023/sourcecode/lab/lab2/ParcelableExampleKotlin/app/src/main/java/pl/edu/uwr/pum/parcelableexamplekotlin/SecondActivity.kt
package pl.edu.uwr.pum.parcelableexamplekotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView

class SecondActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        val prop = intent.getParcelableExtra<Properties>(EXTRA_KEY)
        val serProp = intent.getSerializableExtra(EXTRA_SERIALIZABLE) as SerializableProperties
        val parProp = intent.getParcelableExtra<ParcelizeProperties>(EXTRA_PARCELIZE)

        if (prop != null && parProp != null)
            ((prop.a + prop.b).toString() + " " + prop.c + "----" +
                    serProp.a + serProp.b + " " + serProp.c + "----\n" +
                    ( parProp.a + parProp.b).toString() + " " + parProp.c)
                .also { findViewById<TextView>(R.id.textView).text = it }
    }
}